Header header
string location
bool with_person